src/services/utilServices/SitesService.js

```diff
@@ -74,6 +100,12 @@ class SitesService {
             isPrivate,
           }
         })
+        .filter(
+          (repoData) =>
+            isAdminUser ||
+            !isEmailUser ||
+            retrievedSitesByEmail.includes(repoData.repoName)
+        )
```

Comment:
could i ask why we're doing a `filter` operation here rather than just an early return above, when `isAdminUser` and `isEmailUser` is known? 

it seems like it'll still return an empty array and it would be better for readability too

Reply:
Hmm I think this part isn't very clear - the idea was for all sites retrieved to pass through if the user was an admin user, or using their own github token (since that only retrieves repos they have access to). I've refactored the code to make this clearer in https://github.com/isomerpages/isomercms-backend/pull/480/commits/86cdc315049dfd6dead583c715fefe02329822ce

Reply:
i've made a comment below wrt this - i think it reads better now but the underlying assumption (that if they're github users, then they are using their own token, which will retrieve your owned sites and we can omit the `filter` on our end) should be documented here!

Reply:
Makes sense, added comment in 85aa065